test/unit/teacher_set_test.rb

```diff
@@ -182,13 +193,293 @@ class TeacherSetTest < ActiveSupport::TestCase
+      @mintest_mock3.expect(:call, teacher_set, [ts_items_info])
+      @mintest_mock4.expect(:call, teacher_set)
+      @mintest_mock5.expect(:call, subjects_arr, ['650'])
+      @mintest_mock6.expect(:call, teacher_set, [subjects_arr])
+      @mintest_mock7.expect(:call, "Learning set", ['500', true])
+      @mintest_mock8.expect(:call, teacher_set, ["Learning set"])
+      @mintest_mock9.expect(:call, teacher_set, [SIERRA_USER["data"][0]])
+      @mintest_mock10.expect(:call, teacher_set)
+
+      TeacherSet.stub :initialize_teacher_set, teacher_set, [7899158] do
+        teacher_set.stub :get_items_info_from_bibs_service, ts_items_info, [7899158] do
+          teacher_set.stub :update_teacher_set_attributes_from_bib_request, @mintest_mock3 do
+            teacher_set.stub :clean_primary_subject, @mintest_mock4 do
+              teacher_set.stub :all_var_fields, @mintest_mock5 do
+                teacher_set.stub :update_subjects_via_api, @mintest_mock6 do
+                  teacher_set.stub :var_field_data, @mintest_mock7 do
+                    teacher_set.stub :update_notes, @mintest_mock8 do
+                      teacher_set.stub :update_included_book_list, @mintest_mock9 do
+                        teacher_set.stub :create_or_update_teacherset_document_in_es, @mintest_mock10 do
+                          resp = TeacherSet.create_or_update_teacher_set_data(SIERRA_USER["data"][0])
+                        end
+                      end
+                    end
+                  end
+                end
+              end
+            end
+          end
+        end
+      end
+      assert_equal(teacher_set.bnumber, resp.bnumber)
+      @mintest_mock3.verify
+      @mintest_mock4.verify
+      @mintest_mock5.verify
+      @mintest_mock6.verify
+      @mintest_mock7.verify
+      @mintest_mock8.verify
+      @mintest_mock9.verify
+      @mintest_mock10.verify
+    end
+  end
+
+
+  describe 'delete teacher_set record from db and ES' do
+    it 'test delete_teacher_set_method' do
+      bib_id = '1234'
+      teacher_set = TeacherSet.new(bnumber: "b#{bib_id}")
+      @mintest_mock1.expect(:call, teacher_set, [bib_id])
+  
+
+      resp = nil
+      TeacherSet.stub :get_teacher_set_by_bnumber, @mintest_mock1 do
+        teacher_set.stub :destroy, teacher_set do
+          teacher_set.stub :delete_teacherset_record_from_es, teacher_set do
+            resp = TeacherSet.delete_teacher_set(bib_id)
+          end
+        end
+      end
+      assert_equal(teacher_set.bnumber, resp.bnumber)
+      @mintest_mock1.verify
+    end
+
+    it 'test bib record not found exception' do
+      bib_id = '1234'
+      resp = nil
+      teacher_set = TeacherSet.new(bnumber: "")
+      @mintest_mock1.expect(:call, [], [bib_id])
+      resp = assert_raises(BibRecordNotFoundException) do
+        TeacherSet.stub :get_teacher_set_by_bnumber, @mintest_mock1 do
+          teacher_set.stub :delete_teacherset_record_from_es, teacher_set do
```

Comment:
Does this need to be stubbed? It doesn't look like it should be called according to https://github.com/NYPL/MyLibraryNYCApp/blob/044b9dfdb51f76aae2971dd2b5f885752bc83edf/app/models/teacher_set.rb#L244

Reply:
It feels like we can test `assert_raises(BibRecordNotFoundException)` on a nonexistent bib_id without mocking or stubbing anything.